spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitBatchIntegrationTests.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.BatchingRabbitTemplate;
+import org.springframework.amqp.rabbit.core.support.SimpleBatchingStrategy;
+import org.springframework.amqp.rabbit.junit.RabbitAvailable;
+import org.springframework.amqp.rabbit.junit.RabbitAvailableCondition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * @author Gary Russell
+ * @since 2.2
+ *
+ */
+@SpringJUnitConfig
+@DirtiesContext
+@RabbitAvailable(queues = { "batch.1", "batch.2" })
+public class EnableRabbitBatchIntegrationTests {
+
+	@Autowired
+	private BatchingRabbitTemplate template;
+
+	@Autowired
+	private Listener listener;
+
+	@Test
+	public void simpleList() throws InterruptedException {
+		this.template.convertAndSend("batch.1", new Foo("foo"));
+		this.template.convertAndSend("batch.1", new Foo("bar"));
+		assertThat(this.listener.foosLatch.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(this.listener.foos.get(0).getBar()).isEqualTo("foo");
+		assertThat(this.listener.foos.get(1).getBar()).isEqualTo("bar");
+	}
+
+	@Test
+	public void messageList() throws InterruptedException {
+		this.template.convertAndSend("batch.2", new Foo("foo"));
+		this.template.convertAndSend("batch.2", new Foo("bar"));
+		assertThat(this.listener.fooMessagesLatch.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(this.listener.fooMessages.get(0).getPayload().getBar()).isEqualTo("foo");
+		assertThat(this.listener.fooMessages.get(1).getPayload().getBar()).isEqualTo("bar");
+	}
+
+	@Configuration
+	@EnableRabbit
+	public static class Config {
+
+		@Bean
+		public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory() {
+			SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
+			factory.setConnectionFactory(connectionFactory());
+			factory.setBatchListener(true);
+			return factory;
+		}
+
+		@Bean
+		public BatchingRabbitTemplate template() {
+			return new BatchingRabbitTemplate(connectionFactory(), new SimpleBatchingStrategy(2, 10_000, 10_000L),
+					scheduler());
+		}
+
+		@Bean
+		public ConnectionFactory connectionFactory() {
+			return new CachingConnectionFactory(RabbitAvailableCondition.getBrokerRunning().getConnectionFactory());
+		}
+
+		@Bean
+		public TaskScheduler scheduler() {
+			return new ThreadPoolTaskScheduler();
+		}
+
+		@Bean
+		public Listener listener() {
+			return new Listener();
+		}
+
+	}
+
+	public static class Listener {
+
+		List<Foo> foos;
```

Comment:
Time to ditch `foos` and `bars`? 😉

Reply:
While I understand Jay's point about docs, I don't think we have to change all of our tests.